extensions/arc/runtime/src/main/java/io/quarkus/arc/properties/IfBuildProperty.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.arc.properties;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * When applied to a bean class or producer method (or field), the bean will only be enabled
+ * if the Quarkus build time property matches the provided value.
+ * By default, the bean is not enabled when the build time property is not defined at all, but this behavior is configurable
+ * via the {#code enableIfMissing} property.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE, ElementType.FIELD })
+public @interface IfBuildProperty {
+
+    /**
+     * Name of the build time property to check
+     */
+    String name();
+
+    /**
+     * Expected {@code String} value of the build time property (specified by {@code name}) if the bean is to be enabled
+     */
+    String stringValue();
```

Comment:
Just curious, would it make sense to use a regular expression here? eg. `v.*`

Reply:
I'm ambivalent because perhpaps that would lead to suprising results for the user. I'll let @manovotn and @mkouba decide.

Reply:
Well, regular expressions are powerful but tricky. I'd prefer to keep it simple for now.